quic/client/handshake/FizzClientHandshake.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+#pragma once
+
+#include <quic/client/handshake/ClientHandshake.h>
+
+namespace quic {
+
+class FizzClientContext;
+
+class FizzClientHandshake : public ClientHandshake {
+ public:
+  FizzClientHandshake(
+      QuicCryptoState& cryptoState,
+      std::shared_ptr<FizzClientContext> fizzContext);
+
+ private:
+  std::shared_ptr<FizzClientContext> fizzContext_;
```

Comment:
Is there a reason this is a `shared_ptr` actually? I don't think there's a situation where the `FizzClientHandshake` can exist after the `ClientHandshakeFactory` in the `QuicClientConnectionState` has been destroyed, right? It seems it could just be a `weak_ptr` or even a raw pointer.

Reply:
Because the handshake use delayed destruction, this is not a given.